### PR TITLE
Dice tower

### PR DESCRIPTION
**Pull request by frenche6**
Added a dice tower, which tracks rolls and rolling history.

### CODE REVIEW

File: GameAssistant/Interfaces/IDiceResolver.cs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+﻿using System.Collections.Generic;
+using GameAssistant.Models;
+
+namespace GameAssistant.Interfaces
+{
+    public interface IDiceResolver<T>
+    {
+        DiceResolution<T> ModifyDice(List<BaseDie<T>> dice);
```
Comment (Joefficial) [inline GameAssistant/Interfaces/IDiceResolver.cs:14]: can we get a /// comment here

File: GameAssistant.UnitTest/DiceTowerTests.cs [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        }
+
+        [Fact]
+        public void AddDie_AddsDieSuccessfully()
+        {
+            //var modifiedResults = new List<int>() { 3 };
+
+            //var diceResolution = new DiceResolution<int>(modifiedResults, "3");
+            var diceResolverMock = new Mock<IDiceResolver<int>>();
+            //diceResolverMock.Setup(x => x.ModifyDice(It.IsAny<List<BaseDie<int>>>())).Returns(diceResolution);
```
Comment (carlingkirk) [inline GameAssistant.UnitTest/DiceTowerTests.cs]: Comments

File: GameAssistant/Models/RollHistory.cs [hunk 3]
```diff
@@ -0,0 +1,13 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace GameAssistant.Models
+{
+    public class RollHistory<T>
```
Comment (Joefficial) [inline GameAssistant/Models/RollHistory.cs:6]: summary comments here?

File: GameAssistant.UnitTest/DiceTowerTests.cs [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace GameAssistant.UnitTest
+{
+    public class DiceTowerTests
+    {
+        
+
```
Comment (carlingkirk) [inline GameAssistant.UnitTest/DiceTowerTests.cs]: Too many spaces

File: GameAssistant/Interfaces/IDiceTower.cs [hunk 5]
```diff
@@ -0,0 +1,41 @@
+﻿using System.Collections.Generic;
+using GameAssistant.Models;
+
+namespace GameAssistant.Interfaces
+{
+    public interface IDiceTower<T>
+    {
+        /// <summary>
+        /// Adds a single die to the dice bag
```
Comment (carlingkirk) [inline GameAssistant/Interfaces/IDiceTower.cs]: dice tower?

File: GameAssistant/Services/DiceResolver.cs [hunk 6]
```diff
@@ -0,0 +1,16 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using GameAssistant.Interfaces;
+using GameAssistant.Models;
+
+namespace GameAssistant.Services
+{
+    public class DiceResolver<T> : IDiceResolver<T>
```
Comment (Joefficial) [inline GameAssistant/Services/DiceResolver.cs:8]: is this just an example provider to show use?

File: GameAssistant/Services/DieRollerService.cs [hunk 7]
```diff
@@ -1,6 +1,9 @@
-﻿namespace GameAssistant.Services
+﻿using GameAssistant.Models;
+
+namespace GameAssistant.Services
 {
-    class DieRollerService
+    public class DieRollerService<T>
```
Comment (carlingkirk) [inline GameAssistant/Services/DieRollerService.cs:5]: Delete?

Comment (Joefficial) [inline GameAssistant/Services/DieRollerService.cs:5]: is this used for anything? can it be removed?

File: GameAssistant.UnitTest/DiceTowerTests.cs (hunk 2 above)
Comment (frenche6) [reply GameAssistant.UnitTest/DiceTowerTests.cs]: Removed

File: GameAssistant.UnitTest/DiceTowerTests.cs (hunk 4 above)
Comment (frenche6) [reply GameAssistant.UnitTest/DiceTowerTests.cs]: Removed

File: GameAssistant/Interfaces/IDiceTower.cs (hunk 5 above)
Comment (frenche6) [reply GameAssistant/Interfaces/IDiceTower.cs]: Updated comments

File: GameAssistant/Services/DieRollerService.cs (hunk 7 above)
Comment (frenche6) [reply GameAssistant/Services/DieRollerService.cs:5]: Leaving for now, may delete

File: GameAssistant/Interfaces/IDiceResolver.cs (hunk 1 above)
Comment (frenche6) [reply GameAssistant/Interfaces/IDiceResolver.cs:14]: Added comments

File: GameAssistant/Models/RollHistory.cs (hunk 3 above)
Comment (frenche6) [reply GameAssistant/Models/RollHistory.cs:6]: Added comments

File: GameAssistant/Services/DieRollerService.cs (hunk 7 above)
Comment (frenche6) [reply GameAssistant/Services/DieRollerService.cs:5]: Will probably remove later

**Review by carlingkirk — APPROVED** [review]

**Pull request merged**